intrastat_product/report/intrastat_product_report_xls.py

```diff
@@ -0,0 +1,295 @@
+                    'type': 'string',
+                    'value': self._('Intrastat Region'),
+                },
+                'line': {
+                    'value': self._render("line.region_id.name or ''"),
+                },
+                'width': 28,
+            },
+            'invoice': {
+                'header': {
+                    'type': 'string',
+                    'value': self._('Invoice'),
+                },
+                'line': {
+                    'value': self._render("line.invoice_id.number"),
+                },
+                'width': 18,
+            },
+        }
+        template.update(
+            self.env['intrastat.product.declaration']._xls_template())
+
+        return template
+
+    def getObjects(self, cr, uid, ids, context):
+        """
+        Adapt logic since localization modules have other
+        object tables than the object on which this report is defined.
+        """
+        active_model = self.env.context['active_model']
+        active_id = self.env.context['active_id']
+        declaration = self.env[active_model].browse(active_id)
+        return declaration
+
+    def _get_ws_params(self, wb, data, declaration):
+        template = self._get_template()
+        if self.env.context.get('computation_lines'):
+            wl = declaration._xls_computation_line_fields()
+            report = 'computation'
+        else:
+            wl = declaration._xls_declaration_line_fields()
+            report = 'declaration'
+
+        title = self._get_title(declaration, report, format='normal')
+        title_short = self._get_title(declaration, report, format='short')
+        sheet_name = title_short[:31].replace('/', '-')
+
+        params = {
+            'ws_name': sheet_name,
+            'generate_ws_method': '_intrastat_report',
+            'title': title,
+            'wanted_list': wl,
+            'col_specs': template,
+        }
+        return [params]
+
+    def _get_title(self, declaration, report, format='normal'):
+        title = declaration.year_month
+        if format == 'normal':
+            if report == 'computation':
+                title += ' : ' + _('Computation Lines')
+            else:
+                title += ' : ' + _('Declaration Lines')
+        return title
+
+    def _report_title(self, ws, row_pos, ws_params, data, declaration):
+        return self._write_ws_title(ws, row_pos, ws_params)
+
+    def _empty_report(self, ws, row_pos, ws_params, data, declaration,
+                      report):
+        if report == 'computation':
+            lines = _('Computation Lines')
+        else:
+            lines = _('Declaration Lines')
+        no_entries = _("No") + " " + lines + " " + _("for period %s") \
+            % declaration.year_month
+        ws.write_string(row_pos, 0, no_entries, self.format_left_bold)
+
+    def _intrastat_report(self, workbook, ws, ws_params, data, declaration):
+
+        ws.set_portrait()
```

Comment:
A landscape worksheet would be more readable, particularly for the computation lines spreadsheet.